lib/msal-browser/src/broker/client/BrokerClientApplication.ts

```diff
@@ -3,34 +3,54 @@
-import { BrokerRedirectResponse } from "./BrokerRedirectResponse";
-import { RedirectRequest } from "../request/RedirectRequest";
-import { BrokerAuthResponse } from "./BrokerAuthResponse";
-import { ClientApplication } from "../app/ClientApplication";
-import { PopupRequest } from "../request/PopupRequest";
-import { BrokerHandleRedirectRequest } from "./BrokerHandleRedirectRequest";
-import { SsoSilentRequest } from "../request/SsoSilentRequest";
-import { BrowserStateObject } from "../utils/BrowserProtocolUtils";
+import { version } from "../../../package.json";
+import { BrokerAuthenticationResult, ServerTelemetryManager, AuthorizationCodeClient, BrokerAuthorizationCodeClient, BrokerRefreshTokenClient, RefreshTokenClient, StringUtils, AuthorizationUrlRequest, PersistentCacheKeys, IdToken, ProtocolUtils, ResponseMode, ScopeSet, AccountInfo, AuthenticationScheme } from "@azure/msal-common";
+import { BrokerMessage } from "../msg/BrokerMessage";
+import { BrokerMessageType, InteractionType } from "../../utils/BrowserConstants";
+import { Configuration } from "../../config/Configuration";
+import { BrokerHandshakeRequest } from "../msg/req/BrokerHandshakeRequest";
+import { BrokerHandshakeResponse } from "../msg/resp/BrokerHandshakeResponse";
+import { BrokerAuthRequest } from "../msg/req/BrokerAuthRequest";
+import { BrokerRedirectResponse } from "../msg/resp/BrokerRedirectResponse";
+import { RedirectRequest } from "../../request/RedirectRequest";
+import { BrokerAuthResponse } from "../msg/resp/BrokerAuthResponse";
+import { ClientApplication } from "../../app/ClientApplication";
+import { PopupRequest } from "../../request/PopupRequest";
+import { BrokerHandleRedirectRequest } from "../msg/req/BrokerHandleRedirectRequest";
+import { SsoSilentRequest } from "../../request/SsoSilentRequest";
+import { BrowserStateObject } from "../../utils/BrowserProtocolUtils";
+import { BrokerSilentRequest } from "../request/BrokerSilentRequest";
+import { BrokerAuthError } from "../../error/BrokerAuthError";
+import { BrokerPopupRequest } from "../request/BrokerPopupRequest";
+import { BrokerRedirectRequest } from "../request/BrokerRedirectRequst";
 
 /**
  * Broker Application class to manage brokered requests.
  */
 export class BrokerClientApplication extends ClientApplication {
 
     private cachedBrokerResponse: Promise<BrokerAuthenticationResult>;
+    private brokerAccount: AccountInfo;
 
     constructor(configuration: Configuration) {
         super(configuration);
     }
 
+    /**
+     * 
+     */
+    async handleRedirectPromise(): Promise<BrokerAuthenticationResult | null> {
+        this.cachedBrokerResponse = super.handleRedirectPromise() as Promise<BrokerAuthenticationResult>;
+        const cachedResponse = (await this.cachedBrokerResponse);
+        // TODO: What to do in cases of multi-account in broker?
+        if (cachedResponse) {
+            this.brokerAccount = cachedResponse.account;
+            // return cachedResponse;
```

Comment:
Should this be uncommented?

Reply:
We only want to return here if the `cachedResponse` is for the broker. I will update this in a following PR.

Reply:
I have updated the comment.